src/constants/dbTables.js

```diff
@@ -0,0 +1,114 @@
+module.exports = {
+  roleSchema: [
+    {
+      name: 'name',
+      type: 'TEXT',
+      primaryKey: false,
+      notNull: true,
+      unique: false,
+    },
+  ],
+
+  userSchema: [
+    {
+      name: 'username',
+      type: 'TEXT',
+      primaryKey: false,
+      notNull: true,
+      unique: false,
+    },
+    {
+      name: 'hashed_password',
+      type: 'TEXT',
+      primaryKey: false,
+      notNull: true,
+      unique: false,
+    },
+    {
+      name: 'salt',
+      type: 'NUMERIC',
+      primaryKey: false,
+      notNull: true,
+      unique: false,
+    },
+
+    {
+      name: 'is_superuser',
+      type: 'BOOLEAN',
+      primaryKey: false,
+      notNull: true,
+      unique: false,
+    },
+  ],
+
```

Comment:
Hello @AbegaM - for `_roles_permissions` there is a dual constraint on `role` and `tableName`:
![image](https://github.com/thevahidal/soul/assets/1108513/5dc9c9ee-1519-4e0c-9aac-fc9a28da4b66)

Reply:
@IanMayo Good point, i have moved the `createTable` function that was created by @thevahidal to the `services` folder and i see that it doesn't implement a logic to process fields with multiple `unique constraints`, i will modify the `table service` so it can handle this logic